jobs/intg-test-resources/ballerina-intg-test-cfn.yaml

```diff
@@ -0,0 +1,404 @@
+          tar -xvzf apache-maven-${MavenVersion}-bin.tar.gz
+          ln -fs apache-maven-${MavenVersion} maven
+          echo 'export MAVEN_OPTS="-Xmx2048m -Xms256m"' >> /etc/environment
+          echo 'export M3_HOME=/opt/wso2/workspace/maven' >> /etc/environment
+          echo PATH=/opt/wso2/workspace/maven/bin/:$PATH >> /etc/environment
+
+          source /etc/environment
+
+          cat /dev/null > ~/.bash_history && history -c
+          ${CustomUserData}
+          /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackName} --resource WSO2UnixInstance --region ${AWS::Region}
+      Tags:
+        - Key: Name
+          Value: wso2-apim-intg-node
+      SecurityGroups:
+        - !Ref WSO2InstanceSecurityGroup
+    Metadata:
+      'AWS::CloudFormation::Designer':
+        id: 8f512cce-69ee-45de-87ab-5aedd2a93d50
+      "AWS::CloudFormation::Init" : {
+          "config" : {
+              "packages" : {
+                  "apt" : {
+                    }
+                },
+               "sources" : {
+                },
+               "files" : {
+               }
+          }
+       }
+  WSO2WindowsInstance:
+      Type: 'AWS::EC2::Instance'
+      Condition: 'IsWindows'
+      Properties:
+        DisableApiTermination: 'false'
+        InstanceInitiatedShutdownBehavior: stop
+        ImageId: !Ref AMI
+        InstanceType: 'm3.2xlarge'
+        KeyName: !Ref EC2KeyPair
+        Monitoring: 'false'
+        "BlockDeviceMappings" : [
+           {
+              "DeviceName" : "/dev/sda1",
+              "Ebs" : { "VolumeSize" : "50" }
+           },
+        ]
+        UserData:
+          !Base64
+          'Fn::Sub': |
+            <powershell>
+            [Environment]::SetEnvironmentVariable("JAVA_HOME", "$Env:${JDK}", "Machine")
+
+            wget https://archive.apache.org/dist/maven/maven-3/${MavenVersion}/binaries/apache-maven-${MavenVersion}-bin.zip -OutFile "maven.zip"
+            Expand-Archive maven.zip -DestinationPath c:\\testgrid\\workspace
+
+            [Environment]::SetEnvironmentVariable("M2_HOME", "c:\\testgrid\\workspace\\apache-maven-${MavenVersion}", "Machine")
+            [Environment]::SetEnvironmentVariable("MAVEN_HOME", "c:\\testgrid\\workspace\\apache-maven-${MavenVersion}", "Machine")
+
+            $Old_Path=(Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name Path).Path
+            Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH -Value ($Old_Path += ';c:\\testgrid\\workspace\\apache-maven-${MavenVersion}\\bin') -Verbose -PassThru|fl
+            cd C:\"Program Files"\telegraf
+            curl http://169.254.169.254/latest/meta-data/instance-id -o instance_id.txt
+            ${CustomUserData}
+            .\telegraf.exe  --service install > service.log
```

Comment:
@tharindu1992 Why do we have testgrid specific logic in the cfn script? All testgrid specific logic need to go into custom user data!?

Reply:
I will add all of these to Custom User Data.